libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -32,23 +33,20 @@ public class MicrosoftAppCredentials : ServiceClientCredentials
         /// </summary>
         public const string MicrosoftAppPasswordKey = "MicrosoftAppPassword";
 
-        /// <summary>
-        /// The token refresh code uses this client. Ideally, this would be passed in or set via a DI system to 
-        /// allow developer control over behavior / headers / timeouts and such. Unfortunately this is buried
-        /// pretty deep, the static solution used here is much cleaner. If this becomes an issue we could
-        /// consider circling back and exposing developer control over this HttpClient. 
-        /// </summary>
-        private static readonly HttpClient _httpClient = new HttpClient();
+        private static HttpClient _httpClient = new HttpClient();
 
-        private static object _trustedHostNamesSync = new object();
         private static readonly IDictionary<string, DateTime> _trustedHostNames = new Dictionary<string, DateTime>()
         {
-            { "state.botframework.com", DateTime.MaxValue },
-            { "api.botframework.com", DateTime.MaxValue}
+            // { "state.botframework.com", DateTime.MaxValue }, // deprecated state api
+            { "api.botframework.com", DateTime.MaxValue},       // bot connector API
+            { "token.botframework.com", DateTime.MaxValue }     // oauth token endpoint
         };
 
-        private static object _cacheSync = new object();
-        protected static readonly IDictionary<string, OAuthResponse> _cache = new Dictionary<string, OAuthResponse>();
+
+        protected static readonly Dictionary<string, Task<OAuthResponse>> tokenTaskCache = new Dictionary<string, Task<OAuthResponse>>();
```

Comment:
So, across just the static fields in this class there is a difference of naming conventions where `_trustedHostNames` has the `_` which is usually reserved for instance members and then the others don't. Suggest following the .NET guideline of using `TitleCase` for statics for all of them.